pkg/liqo-controller-manager/offloadingStatus-controller/offloadingstatus_controller.go

```diff
@@ -0,0 +1,153 @@
+			// The NamespaceOffloading is deleted only when there are no more remoteNamespaceCondition and
+			// the deletion timestamp is set.
+			for key := range noff.Status.RemoteNamespacesConditions {
+				delete(noff.Status.RemoteNamespacesConditions, key)
+			}
+			klog.Infof("NamespaceOffloading, in the namespace '%s', ready to be deleted", noff.Namespace)
+		}
+	case ready+notReady == 0:
+		noff.Status.OffloadingPhase = offv1alpha1.NoClusterSelectedOffloadingPhaseType
+	case ready == 0:
+		noff.Status.OffloadingPhase = offv1alpha1.AllFailedOffloadingPhaseType
+	case notReady == 0:
+		noff.Status.OffloadingPhase = offv1alpha1.ReadyOffloadingPhaseType
+	default:
+		noff.Status.OffloadingPhase = offv1alpha1.SomeFailedOffloadingPhaseType
+	}
+
+	klog.Infof("The OffloadingStatus for NamespaceOffloading in the namespace '%s' is set to '%s'",
+		noff.Namespace, noff.Status.OffloadingPhase)
+}
+
+// Reconcile sets the NamespaceOffloading Status checking the actual status of all remote Namespace.
+// This reconcile is performed every RequeueTime.
+func (r *OffloadingStatusReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	namespaceOffloading := &offv1alpha1.NamespaceOffloading{}
+	if err := r.Get(ctx, types.NamespacedName{
+		Namespace: req.Namespace,
+		Name:      liqoconst.DefaultNamespaceOffloadingName,
+	}, namespaceOffloading); err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.Infof("There is no NamespaceOffloading resource in Namespace '%s'", req.Namespace)
+			return ctrl.Result{}, nil
+		}
+		klog.Errorf("%s --> Unable to get namespaceOffloading for the namespace '%s'", err, req.Namespace)
+		return ctrl.Result{}, err
+	}
+
+	// Get all NamespaceMaps in the cluster
+	namespaceMapList := &mapsv1alpha1.NamespaceMapList{}
+	if err := r.List(ctx, namespaceMapList); err != nil {
+		klog.Errorf("%s -> unable to get NamespaceMaps", err)
+		return ctrl.Result{}, err
+	}
+
+	if len(namespaceMapList.Items) == 0 {
+		klog.Info("No NamespaceMaps in the cluster at the moment")
+		return ctrl.Result{RequeueAfter: r.RequeueTime}, nil
+	}
+
+	patch := namespaceOffloading.DeepCopy()
+
+	setRemoteConditionsForEveryCluster(namespaceOffloading, namespaceMapList)
+
+	setNamespaceOffloadingStatus(namespaceOffloading)
+
+	// Patch the status just one time at the end of the logic.
+	if err := r.Patch(ctx, namespaceOffloading, client.MergeFrom(patch)); err != nil {
+		klog.Errorf("%s --> Unable to Patch NamespaceOffloading in the namespace '%s'",
+			err, namespaceOffloading.Namespace)
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{RequeueAfter: r.RequeueTime}, nil
+}
+
+// checkNamespaceOffloadingStatus calls Reconcile only when a new NamespaceOffloading is created.
+func checkNamespaceOffloadingStatus() predicate.Predicate {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return false
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return false
+		},
+	}
+}
```

Comment:
Two comments:
1) Why are you relying on the requeue mechanism instead of watching the namespace maps and react when a modification occurs?
2) I would double check what happens if the controller is restarted. Do you get a create event which restores the requeue, or an update one which is filtered by the predicate?

Reply:
1. For the moment we choose this implementation...
2. I have checked it should get a create event which restores the requeue